src/net_processing.cpp

```diff
@@ -1541,11 +1541,11 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
             if (mi != mapRelay.end()) {
                 connman->PushMessage(pfrom, msgMaker.Make(nSendFlags, NetMsgType::TX, *mi->second));
                 push = true;
-            } else if (pfrom->m_tx_relay->timeLastMempoolReq) {
+            } else if (pfrom->m_tx_relay->m_last_mempool_req.load().count()) {
```

Comment:
why not use `count_seconds` here, if it is preferred to inline `.count()` (according to the comment)?

Reply:
The member uses a magic value of 0 to indicate that the mempool was never requested. So this check isn't about time (in a specific type like seconds), but more about checking that any value exists.

Reply:
Hmm, okay, I guess we'd really want a `optional` here some day instead of a magic value, but no need to do that here.

Reply:
Sorry to chime in late -- would it make sense to use the `std::chrono::duration::zero` value here?  I found `count()` to be unintuitive when I was looking at related code just now (but this `std::chrono` stuff is new to me so I'm still figuring out how to use it).

Reply:
Comparing it to zero would also work, but still be odd, as the code is written as if the member were an `optional`.

I think we can just stop treating it as `optional` (always assume the last mempool request time for peers was in year 1970). But that can be done in a follow up pull request.